### DISCUSSION TITLE
Grafana Unable to login

### DISCUSSION THREAD

**Discussion by yomaoni** [Security-Onion-Solutions/securityonion#8741] [Unsupported Versions]
Trying to login to Grafana to build some alerts but having some trouble.  

Login with admin (not Admin) and the password i got from salt-call pillar.get secrets 

Keep getting Invalid username or password

not sure what I am doing wrong

Thanks

**Comment by reyesj2** [comment]
Try copying and pasting the password. If you already are copying and pasting try typing the password manually. You may be copying one characters too many.

**Reply by yomaoni** [reply]
ya tried that, no luck. Was wondering if it is possible to reset the password.

**Comment by cm-ops** [comment]
`sudo docker exec -it so-grafana  grafana-cli --homepath "/usr/share/grafana"  admin reset-admin-password admin`

Run the above command, this will reset the admin password to admin. Then open Grafana and try to login with admin:admin, it will ask you to change the password. Copy the password from secrets and reset the password to what is in secrets.

**Reply by yomaoni** [reply]
worked like a charm. Thanks a ton

**Comment by swannman** [comment]
This happened to me as well today. It looks like multiple discussion topics have been raised on the same issue (8741, 8206, 7436), is it possible that there is a bug here?

**Reply by dougburks** [reply]
@swannman I just tested multiple deployments and was able to login to Grafana as `admin` using the instructions at https://docs.securityonion.net/en/2.3/grafana.html#accounts, so this appears to be working properly.

Were you able to use the instructions above to reset your Grafana admin password back to the stored password?

**Reply by swannman** [reply]
Hi @dougburks, I was able to repro this issue:

1. `sudo bash so-setup-network`. Select `standalone` and accept the default settings.
2. `sudo salt-call pillar.get secrets`. This returns `grafana_admin: password1`. 
   - I verified that `password1` was correct for the `admin` user in Grafana.
4. `sudo bash so-setup-network` again. Select `standalone` and accept the default settings.
5. `sudo salt-call pillar.get secrets`. This returns `grafana_admin: password2`.
   - However, the `admin` user in Grafana was not updated and is still set to `password1`.

Is this expected? This was on 2.3.170.

**Reply by dougburks** [reply]
I've created the following issue:
https://github.com/Security-Onion-Solutions/securityonion/issues/8949

In the meantime, the workaround above should work to reset the Grafana admin password:
https://github.com/Security-Onion-Solutions/securityonion/discussions/8741#discussioncomment-3657071
